### PR TITLE
feat: return document name

### PR DESCRIPTION
**Pull request by Montel**

### CODE REVIEW

File: src/core/document.h [hunk 1]
```diff
@@ -50,6 +50,8 @@ class Document : public QObject
 
     Type type() const;
 
+    Q_INVOKABLE const QString documentName() const;
```
Comment (narnaud) [inline src/core/document.h:53]: This should be a property, like fileName, not an invokable method.

File: src/core/document.h [hunk 2]
```diff
@@ -21,6 +21,7 @@ class Document : public QObject
 {
     Q_OBJECT
     Q_PROPERTY(QString fileName READ fileName WRITE setFileName NOTIFY fileNameChanged)
+    Q_PROPERTY(QString documentName READ documentName NOTIFY documentNameChanged)
```
Comment (narnaud) [inline src/core/document.h:24]: Reuse fileNameChanged signal: no need for a second signal, as the 2 are linked.

File: src/core/document.cpp [hunk 3]
```diff
@@ -89,6 +89,15 @@ Document::Type Document::type() const
     return m_type;
 }
 
+const QString Document::documentName() const
```
Comment (narnaud) [inline src/core/document.cpp:92]: Missing the property documentation (see at the top of the C++ file)

**Comment by Montel** [comment]
I don't understand why it create merge commit...
I don't know how to remove it.

**Pull request closed without merging**